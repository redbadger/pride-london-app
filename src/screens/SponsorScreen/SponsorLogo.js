// @flow
import React from "react";
import { View, StyleSheet, Image } from "react-native";
import type { SponsorLevel } from "../../data/sponsor";

type Props = {
  sponsorName: string,
  sponsorLogoUrl: string,
  sponsorLevel: SponsorLevel,
  sponsorLogoSize: { width: number, height: number }
};

const SponsorLogo = ({
  sponsorName,
  sponsorLogoUrl,
  sponsorLevel,
  sponsorLogoSize
}: Props) => (
  <View
    style={
      sponsorLevel === "Headline" || sponsorLevel === "Gold"
        ? styles.sponsorImageLarge
        : styles.sponsorImageSmall
    }
  >
    <Image
      style={styles.image}
      resizeMode="contain"
      accessibilityLabel={sponsorName}
      source={{ uri: sponsorLogoUrl, ...sponsorLogoSize }}
    />
  </View>
);

const styles = StyleSheet.create({
  sponsorImageLarge: {
    width: "75%",
    height: "48%"
  },
  sponsorImageSmall: {
    width: "80%",
    height: "48%"
  },
  image: {
    maxWidth: "100%",
    maxHeight: "100%"
  }
});

export default SponsorLogo;
