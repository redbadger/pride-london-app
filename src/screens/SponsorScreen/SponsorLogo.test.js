// @flow
import React from "react";
import { shallow } from "enzyme";
import SponsorLogo from "./SponsorLogo";

describe("SponsorLogo Component", () => {
  it("renders correctly with Gold Level", () => {
    const output = shallow(
      <SponsorLogo
        sponsorName="sponsorName"
        sponsorLogoUrl="sponsorLogoUrl"
        sponsorLevel="Gold"
        sponsorLogoSize={{ width: 100, height: 100 }}
      />
    );
    expect(output).toMatchSnapshot();
  });

  it("renders correctly with Silver Level", () => {
    const output = shallow(
      <SponsorLogo
        sponsorName="sponsorName"
        sponsorLogoUrl="sponsorLogoUrl"
        sponsorLevel="Silver"
        sponsorLogoSize={{ width: 100, height: 100 }}
      />
    );
    expect(output).toMatchSnapshot();
  });
});
