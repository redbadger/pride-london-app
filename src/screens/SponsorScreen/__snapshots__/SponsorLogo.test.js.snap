// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorLogo Component renders correctly with Gold Level 1`] = `
<Component
  style={
    Object {
      "height": "48%",
      "width": "75%",
    }
  }
>
  <Image
    accessibilityLabel="sponsorName"
    resizeMode="contain"
    source={
      Object {
        "height": 100,
        "uri": "sponsorLogoUrl",
        "width": 100,
      }
    }
    style={
      Object {
        "maxHeight": "100%",
        "maxWidth": "100%",
      }
    }
  />
</Component>
`;

exports[`SponsorLogo Component renders correctly with Silver Level 1`] = `
<Component
  style={
    Object {
      "height": "48%",
      "width": "80%",
    }
  }
>
  <Image
    accessibilityLabel="sponsorName"
    resizeMode="contain"
    source={
      Object {
        "height": 100,
        "uri": "sponsorLogoUrl",
        "width": 100,
      }
    }
    style={
      Object {
        "maxHeight": "100%",
        "maxWidth": "100%",
      }
    }
  />
</Component>
`;
